plugin/ignite-plugin/src/main/scala/org/finos/vuu/feature/ignite/filter/IgniteSqlFilterClause.scala

```diff
@@ -1,39 +1,91 @@
-    )
-    sql
+    val sql = clauses.map(c => c.toSql(tableDef)).mkString(" AND ")
+    if (clauses.length > 1) s"($sql)" else sql
   }
 }
-case class EqIgniteSqlFilterClause(columnName: String, value: String) extends IgniteSqlFilterClause{
+
+case class EqIgniteSqlFilterClause(columnName: String, value: String) extends IgniteSqlFilterClause {
   override def toSql(tableDef: TableDef): String = {
+    val processedVal = tableDef.columnForName(columnName).dataType match {
+      case CharDataType | StringDataType => quotedString(value)
+      case _ => value
+    }
+    s"$columnName = $processedVal"
+  }
+}
 
-    tableDef.columnForName(columnName).dataType match {
-      case StringDataType => s"$columnName=\'$value\'"
-      case _ => s"$columnName=$value"
+case class NeqIgniteSqlFilterClause(columnName: String, value: String) extends IgniteSqlFilterClause {
+  override def toSql(tableDef: TableDef): String = {
+    val processedVal = tableDef.columnForName(columnName).dataType match {
+      case CharDataType | StringDataType => quotedString(value)
+      case _ => value
     }
-    //s"$columnName=\'$value\'"
+    s"$columnName != $processedVal"
   }
 }
 
 //todo why is number cast double? need to cast back to original type?
-case class GreaterThanIgniteSqlFilterClause(columnName: String, value: Double) extends IgniteSqlFilterClause{
+case class GtIgniteSqlFilterClause(columnName: String, value: Double) extends IgniteSqlFilterClause {
   override def toSql(tableDef: TableDef): String = s"$columnName > $value"
-}
\ No newline at end of file
+}
+
+case class LtIgniteSqlFilterClause(columnName: String, value: Double) extends IgniteSqlFilterClause {
+  override def toSql(tableDef: TableDef): String = s"$columnName < $value"
+}
+
+case class StartsIgniteSqlFilterClause(columnName: String, value: String) extends IgniteSqlFilterClause with StrictLogging {
+  override def toSql(tableDef: TableDef): String =
+    tableDef.columnForName(columnName).dataType match {
+      case StringDataType => s"$columnName LIKE '$value%'"
+      case _ =>
+        logger.error(s"`Starts` clause unsupported for non string column: `$columnName`")
+        EMPTY_SQL
+    }
+}
+
+case class EndsIgniteSqlFilterClause(columnName: String, value: String) extends IgniteSqlFilterClause with StrictLogging {
+  override def toSql(tableDef: TableDef): String =
+    tableDef.columnForName(columnName).dataType match {
+      case StringDataType => s"$columnName LIKE '%$value'"
+      case _ =>
+        logger.error(s"`Ends` clause unsupported for non string column: `$columnName`")
+        EMPTY_SQL
+    }
+}
+
+case class InIgniteSqlFilterClause(columnName: String, values: List[String]) extends IgniteSqlFilterClause with StrictLogging {
+  override def toSql(tableDef: TableDef): String = {
+    val processedValues = tableDef.columnForName(columnName).dataType match {
+      case CharDataType | StringDataType => values.map(quotedString.apply)
```

Comment:
Not sure what is considered better practice but if you want to call it quotedString like everywhere else we could do
values.map(quotedString(_))
Or
values.map(v=>quotedString(value))

Reply:
I like this `values.map(quotedString(_))` better, concise and neat. Updated ✅